astropy/table/_column_mixins.pyx

```diff
@@ -58,7 +55,17 @@ cdef inline object base_getitem(object self, object item, item_getter getitem):
     if (<ndarray>self).ndim > 1 and isinstance(item, INTEGER_TYPES):
         return self.data[item]
 
-    return getitem(self, item)
+    value = getitem(self, item)
+
+    # In Py3+ return a scalar bytes value as utf-8-encoded str
+    if PYV != 2:
+        try:
+            if value.dtype.char == 'S' and not value.shape:
+                value = value.decode('utf-8')
```

Comment:
Inspired by #6117, should the decoding use `errors='replace'` here to avoid `UnicodeDecodeError` for bytes content that is not UTF-8?